server/src/main/java/org/elasticsearch/common/time/DateFormatters.java

```diff
@@ -221,11 +222,15 @@
         .toFormatter(Locale.ROOT));
 
     private static final DateTimeFormatter DATE_FORMATTER = new DateTimeFormatterBuilder()
-        .appendValue(ChronoField.YEAR, 1, 4, SignStyle.NORMAL)
+        .appendValue(ChronoField.YEAR, 1, 5, SignStyle.NORMAL)
```

Comment:
Is this correct with iso 8601? I'm surprised this is only min 1 digit right now, but I also thought it prescribed 4 digits.

Reply:
this is only for non strict parsing (and to retain bwc with the non strict joda formatters).  the strict formatters are only parsing four digits